From 241a8a101b3ba7a3e2a47d443735a17070ece44e Mon Sep 17 00:00:00 2001
From: Scott Wolchok <swolchok@meta.com>
Date: Wed, 15 Jan 2025 11:33:42 -0800
Subject: [PATCH] Fix erroneous at_vreinterpretq_u16_bf16 call (#144883)

Here, `mask` is definitely a `uint16x8_t`, not an `at_bfloat16x8_t`, so we shouldn't be reintepreting it. Candidate fix for #144818 .

Differential Revision: [D68224128](https://our.internmc.facebook.com/intern/diff/D68224128/)

Pull Request resolved: https://github.com/pytorch/pytorch/pull/144883
Approved by: https://github.com/tinglvv, https://github.com/Skylion007, https://github.com/malfet
---
 aten/src/ATen/cpu/vec/vec128/vec128_bfloat16_neon.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aten/src/ATen/cpu/vec/vec128/vec128_bfloat16_neon.h b/aten/src/ATen/cpu/vec/vec128/vec128_bfloat16_neon.h
index e934d8b0179b5..7d594c696f7a6 100644
--- a/aten/src/ATen/cpu/vec/vec128/vec128_bfloat16_neon.h
+++ b/aten/src/ATen/cpu/vec/vec128/vec128_bfloat16_neon.h
@@ -274,7 +274,7 @@ class Vectorized<c10::BFloat16> : public Vectorized16<at_bfloat16x8_t, c10::BFlo
     Vectorized<c10::BFloat16> vec(
         at_vreinterpretq_bf16_u16(
             vbslq_u16(
-                at_vreinterpretq_u16_bf16(mask),
+                mask,
                 at_vreinterpretq_u16_bf16(b.values),
                 at_vreinterpretq_u16_bf16(a.values))));
 
